utils/src/userInput/showInputBox.ts

```diff
@@ -18,29 +18,40 @@ export async function showInputBox(context: IInternalActionContext, options: typ
         const inputBox: InputBox = createInputBox(context, options);
         disposables.push(inputBox);
 
-        let latestValidation: Promise<string | undefined | null> = options.validateInput ? Promise.resolve(options.validateInput(inputBox.value)) : Promise.resolve('');
+        let latestValidateInputTask: Promise<string | undefined | null> = options.validateInput ? Promise.resolve(options.validateInput(inputBox.value)) : Promise.resolve('');
+        let latestAsyncValidationTask: Promise<string | undefined | null> = Promise.resolve('');
         return await new Promise<string>((resolve, reject): void => {
             disposables.push(
                 inputBox.onDidChangeValue(async text => {
+                    let validateInputTask: Promise<string | undefined | null> | undefined;
                     if (options.validateInput) {
-                        const validation: Promise<string | undefined | null> = Promise.resolve(options.validateInput(text));
-                        latestValidation = validation;
-                        const message: string | undefined | null = await validation;
-                        if (validation === latestValidation) {
+                        validateInputTask = Promise.resolve(options.validateInput(text));
+                        latestValidateInputTask = validateInputTask;
+                        const message: string | undefined | null = await validateInputTask;
+                        if (validateInputTask === latestValidateInputTask) {
                             inputBox.validationMessage = message || '';
                         }
                     }
+                    if (options.asyncValidationTask && !inputBox.validationMessage && validateInputTask === latestValidateInputTask) {
+                        // Start loading the promise, but wrap with a short debounce to avoid performing server calls too frequently
+                        latestAsyncValidationTask = inputBoxDebounce(options.asyncValidationTask, text);
```

Comment:
I may be wrong, but I thought we were okay to only call the async validation task `onDidAccept`. That way we won't have to worry about multiple API calls being made repeatedly.

The example was whenever you register a new account, they don't check if the username is available until you hit submit, then it will come back and say it's unavailable.

@alexweininger Thoughts?

Reply:
Yeah actually I think I'll change it back to that.  I ended up changing it to the internal debounce because I thought it might have some performance benefits, but now that I'm thinking about it, you're always going to be introducing a delay on the last keystroke now which isn't ideal.  I'll just remove the async queue and do it all during `onDidAccept`. 🤔